src/usr/local/www/services_rfc2136_edit.php

```diff
@@ -186,12 +188,26 @@ function build_if_list() {
 	$pconfig['keyname']
 ))->setHelp('This must match the setting on the DNS server.');
 
+$section->addInput(new Form_Select(
+	'keyalgorithm',
+	'*Key algorithm',
+	$pconfig['keyalgorithm'],
+	array(
```

Comment:
You will need to add some input validation to ensure that the submitted value for keyalgorithm is one from this list, which means it's also easier to define this array once at the start of the file and then use the array to validate and also to build this list.